### ISSUE TITLE
BUG: Obsidian Sync deletes my changes in Excalidraw after a few seconds of starting adding things. 

### ISSUE THREAD

**Issue by ctwhome** [zsviczian/obsidian-excalidraw-plugin#1189]
**Your environment**
SYSTEM INFO:
	Obsidian version: v1.3.5
	Installer version: v1.3.5
	Operating system: Darwin Kernel Version 22.5.0: Thu Jun  8 22:22:20 PDT 2023; root:xnu-8796.121.3~7/RELEASE_ARM64_T6000 22.5.0
	Login status: logged in
	Catalyst license: none
	Insider build toggle: off
	Live preview: on
	Legacy editor: off
	Base theme: dark
	Community theme: none
	Snippets enabled: 8
	Restricted mode: off
	Plugins installed: 35
	Plugins enabled: 21
		1: Settings Search v1.3.7
		2: Trash Explorer v1.1.0
		3: Ozan's Image in Editor Plugin v2.2.3
		4: Style Settings v1.0.6
		5: Update time on edit v1.1.2
		6: Omnisearch v1.14.2
		7: Text Extractor v0.4.6
		8: Simple Embeds v1.14.1
		9: Advanced Tables v0.18.1
		10: Obsidian Memos v1.9.7
		11: Auto Link Title v1.4.1
		12: Daily Notes Viewer v0.2.1
		13: Dataview v0.5.56
		14: Duplicate Line v1.4.1
		15: Editor Syntax Highlight v0.1.3
		16: Excalidraw v1.9.6
		17: Home tab v1.1.2
		18: Templater v1.16.0
		19: Better Command Palette v0.15.0
		20: Quiet Outline v0.3.15
		21: Quick Switcher++ v3.3.0

**Describe the bug**
I purchased Obsidian Sync a few days ago. Since I have it, it seems to be a problem with Excalidraw because every time I am adding modifications something comes from the sync

**To Reproduce**
Enable Obsidian Sync and start adding modifications

**Expected behavior**
you will write text, add shapes, remove them, and after a few seconds, maybe 30, it will be a sort of glitch/refresh that will replace your state from an old one. When this happens and I am typing, the focus of the text will go off and suddenly I can't type anymore. I have to click on the text box again with the mouse to continue typing.

This problem basically makes the plugin unusable because it deletes things I write, interferes with what I am modifying and sometimes even resets the zoom!

**Comment by zsviczian** [comment]
I use Excalidraw and Obsidian with Sync on 5 different devices (Windows 11, Windows 10, MacOS, Android, and iOS). Obsidian sync and Excalidraw work well together for me. This leads me to think that there is probably another root cause resulting in this unfortunate situation.

In fact, Excalidraw has features to support editing in parallel on multiple devices using sync. For example, you can draw on your iPad and add text and other elements on your desktop next to each other. 

Are you by any chance storing Obsidian on iCloud, OneDrive, Google Drive, or similar? If you are using sync, make sure you don't double synchronize with another cloud sync approach. That could lead to a ping-pong of files resulting in this behavior.

**Comment by ctwhome** [comment]
> In fact, Excalidraw has features to support editing in parallel on multiple devices using sync. For example, you can draw on your iPad and add text and other elements on your desktop next to each other.

is this a specific setting to set?

I have tried to see if iCloud or Google Drive was bothering it, moved the folders to different locations, etc.. but nothing happened. Syncing times would take an average of 20 seconds, sometimes more, and rarely 15 seconds (minimum). Maybe that is the Expected behavior, but it is far from using both tablet and computer seamlessly.

I created a new remote vault and only synced between Mac and iOS, and the sync time was the same slow. And the problem with zooming is still there. I thought it was random, but it turns to be that it zooms to fit the screen. 

if during those 20 seconds of sync waiting yo modify anything on the other computer, it will be removed after the sync from the other device kicks in.

**Comment by zsviczian** [comment]
Regarding the autozoom, please check zoom settings. Hopefully, you can find the setting you need. If not, please let me know.
![image](https://github.com/zsviczian/obsidian-excalidraw-plugin/assets/14358394/461eb4df-d991-4439-be1f-f01bdb7bc72e)

Regarding synchronization. You were correct, the feature was broken. At some point in time, Obsidian must have rolled out a change that impacted the sequence in which events are triggered, and the view was updated with data before my synch algorithm was executed. I did not notice this, since recently I've been using my Huion drawing tablet when at the PC, not the iPad... anyway, it is fixed in 1.9.7

Regarding the speed, it is not real-time, but with the feature fixed, it works pretty well, since you do not need to wait for the other synchronization to complete, it will merge into your drawing when it arrives. Speed depends on Obsidian sync speed. Now as I was playing with it, the typical turnaround time was between 5 and 10 seconds. The additional setting that influences the speed is autosave frequency. Too frequent autosave can have two negative impacts:
1) performance
2) sync quota. I've not had an issue with this (I've been using sync for 2 years and I have a lot of drawings), but I had a user reporting running out of sync quota and one reason was the many Excalidraw saved versions.
If you set autosave to infrequent, then you need to pay more attention to how you are closing the app. If you "swipe" obsidian away, or even change task on an iPad, iOS controls when apps are offloaded from memory... this can result in loss of unsaved work. But even on a desktop terminating Obsidian can lead to loss of data if you don't close your drawing first. This is because when you click to close Obsidian, Obsidian does not wait for plugins to terminate.

This is the autosave setting I use:
![image](https://github.com/zsviczian/obsidian-excalidraw-plugin/assets/14358394/b8d7c96e-cfd5-4c8b-89fa-e5b77b073d67)

**Comment by zsviczian** [comment]
fix was released in 1.9.7 just now

**Comment by ctwhome** [comment]
I copied your settings and test drawing at the same time in both places (Mac and Android (galaxy with S-pen)) and works very fine. It is far from real-time indeed, I guess that that is a trade-off for now. The most important things are 1, no data lost! thanks for fixing it, 2. and the zoom is not jumping all around when trying a draw on the tablet, and suddenly the computer goes all over.

**Comment by zsviczian** [comment]
Awesome! Thanks for raising this issue. I am sure others were also struggling with this. Good that you raised it!

**Comment by zsviczian** [comment]
I recommend testing it for yourself. I tried one drive, google drive and github before deciding that Obsidian sync is worth the investment, not because of Excalidraw, but for Obsidian in general. But my usecase is/was different. I use Obsidian on Windows, Android, MacOS, iOS, and in the past also on Linux. All the other approaches had limitations.

As long as you have only one sync service active (i.e. you don't place your files on iCloud and use Obsidian sync) you should be ok. In the end, Obsidian Sync also simply moves files between devices.
